modules/fishing.py

```diff
@@ -80,44 +82,25 @@ async def _catch_fish(ctx):
 
 @bot.command()
 async def fish(ctx: Context):
-
-    new_fisher = True
-
-    for i in range(len(sessions)):
-        if ctx.author.id == sessions[i].id:
-            if not sessions[i].state == NOT_FISHING:
-                await ctx.send(f"{ctx.author.mention} You are already fishing.")
-                return
-
-            fisher = i
-            sessions[i].state = WAITING_FOR_BITE
-            new_fisher = False
-            break
-
-    if new_fisher:
-        fisher = len(sessions)
-        sessions.append(FishingSession)
-        sessions[fisher].id = ctx.author.id
-
-    sessions[fisher].state = WAITING_FOR_BITE
-    await ctx.send(f"{ctx.author.mention} You casts your line...")
-    await _bite_timer(ctx, fisher)
+    fisher = ctx.author.id
+    if fisher not in sessions:
+        sessions[fisher] = FishingState.WAITING_FOR_BITE
+        await ctx.send(f"{ctx.author.mention} You casts your line...")
+        await _bite_timer(ctx, fisher)
+    else:
+        await ctx.send(f"{ctx.author.mention} You are already fishing.")
```

Comment:
Super straightforward here, just 7 lines.

We can efficiently both a) check if the user hasn't fished before and b) get the fisher's state if they have.